spec/pluginmanager/sources_spec.rb

```diff
@@ -0,0 +1,150 @@
+# encoding: utf-8
+require "spec_helper"
+require 'webmock/rspec'
+require "tmpdir"
+require "pluginmanager/sources/http"
+require "pluginmanager/sources/local"
+
+describe LogStash::PluginManager::Sources::HTTP do
+
+  subject   {  described_class.new(uri) }
+
+  let(:uri) { "http://localhost:8080/pack.zip"}
+
+  describe "#exist?" do
+
+    it "should verify if a plugin exist in the repository" do
+      stub_request(:head, uri).and_return(:status => 200)
+      expect(subject.exist?).to eq(true)
+    end
+
+    it "should fail if a plugin does not exist in the repository" do
+      stub_request(:head, uri).and_return(:status => 404)
+      expect(subject.exist?).to eq(false)
+    end
+
+  end
+
+  describe "#fetch" do
+
+    before(:each) do
+      File.delete("pack.zip") if File.exist?("pack.zip")
+    end
+
+    after(:each) do
+      File.delete("pack.zip") if File.exist?("pack.zip")
+    end
+
+    it "should start the download process" do
+      stub_request(:get, uri).and_return(:body => "foo", :status => 404)
+      Dir.mktmpdir do |tmp_dir|
```

Comment:
Every time, I do not use Stud in one of my PR for this kind of behavior, I get asked to use it. So I pass this information around. And is not like I asked you to include a new dependency to the gemfile. Its already in the Logstash project and we maintain it.

For the technical point, this library allow user to define the temporary directory in multiple way  that the default library don't. As a user I would be confused if not all the temporary files where saved at the same place.

https://github.com/jordansissel/ruby-stud/blob/master/lib/stud/temporary.rb#L13